blocks/library/latest-posts/index.js

```diff
@@ -72,22 +75,16 @@ registerBlockType( 'core/latest-posts', {
 		}
 
 		componentWillReceiveProps( nextProps ) {
-			const { postsToShow: postToShowCurrent } = this.props.attributes;
-			const { postsToShow: postToShowNext } = nextProps.attributes;
-			const { setAttributes } = this.props;
-
-			if ( postToShowCurrent === postToShowNext ) {
+			if ( isEqual( this.props.attributes, nextProps.attributes ) ) {
```

Comment:
Just noting that we really want to check that a change in the attributes specific to the fetching of posts are the only ones we want to trigger a new request for; as implemented, a change to the block's alignment, columns, layout will trigger a new request.